### ISSUE TITLE
twisted/test/test_ssl.py shouldn't use setUpClass and tearDownClass

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2743]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2743|
|Type|task|
|Created|2007-07-20 14:39:13Z|

TLSTestCase installs an error ignorer in setUpClass and removes it in tearDownClass.

ignoreErrors is deprecated, so it should stop doing that. :)

I _think_ SSL errors are no longer logged in the way they were when this test was written, anyway, so the setUpClass and tearDownClass can probably just be deleted.  (I tried it, and the tests still pass, but I'm not sure if there was an intermittent aspect here.)  Worst case, setUpClass should be deleted and tearDown should have a flushLoggedErrors call added to it.

<details><summary>Searchable metadata</summary>

```
trac-id__2743 2743
type__task task
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1184942353000000 1184942353000000
changetime__1185626420000000 1185626420000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun removed owner|
|-|-|

twisted.internet.tcp indeed delivers all SSL.Error exceptions to connectionLost instead of logging them, so the ignoreErrors call should no longer be necessary.

This is done in setup-teardown-ssl-2743

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @exarkun|
|-|-|

It looks good, you can merge. You can make the usual cleanups before if you feel it (copyright, unused import, remove future import, and trailing whitespaces).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [20855]) Merge setup-teardown-ssl-2743

Author: exarkun
Reviewer: therve
Fixes [#2743](https://github.com/twisted/twisted/issues/2743)

Remove setUpClass from TLSTestCase, as the errors it was causing to be
ignored are no longer logged by the implementation.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
